docs/validation-model/inheritance-principle.md

```diff
@@ -0,0 +1,63 @@
+# The Inheritance Principle
+
+The [Inheritance Principle] is a core concept in BIDS.
+Its original definition (edited for brevity) was:
+
+> Any metadata file (`.json`, `.bvec`, `.tsv`, etc.) may be defined at any directory level,
+> but no more than one applicable file may be defined at a given level.
+> The values from the top level are inherited by all lower levels
+> unless they are overridden by a file at the lower level. [...]
+> There is no notion of "unsetting" a key/value pair.
+
+Here, "top level" means dataset root, and "lower level" means closer to the data file
+the metadata applies to.
+More recent versions of the specification have made the language more precise at the cost
+of verbosity.
+The core concept remains the same.
+
+The validator uses a "walk back" algorithm to find inherited files:
+
+```python
+def walkBack(file, extension):
+    fileParts = parsePath(file.path)
+    
+    fileTree = file.parent
+    while fileTree:
+        for child in fileTree.children:
+            parts = parsePath(child.path)
+            if (
+                parts.extension == extension
+                and parts.suffix = fileParts.suffix
+                and isSubset(parts.entities, fileParts.entities)
+            ):
+                yield child
+
+        fileTree = fileTree.parent
+```
+
+Using this basis, `loadSidecar` is simply:
```

Comment:
BIDS does not have this concept yet, so the validator does not have it. We would need to define it to know whether `walkBack` is even applicable (augment `sub-01_sessions.tsv`), or if we need to construct one global table (augment `sessions.tsv`).

Reply:
To what `.tsv` files we have that principle applied so far ?  
We do list not only `.json` but also `.bvec` and `.tsv` files in https://github.com/bids-standard/bids-specification/blob/master/src/common-principles.md#the-inheritance-principle

Reply:
Yes, those are the associated files, see a couple lines down from here.

Reply:
Thanks!  I do not see any kind of "inheritance" operation there though like here `| sidecar` -- there it just returns first found instead of "building up" the value.

Reply:
> For [tabular files](https://bids-specification.readthedocs.io/en/stable/common-principles.html#tabular-files) and other simple metadata files (for instance, [bvec / bval files for diffusion MRI](https://bids-specification.readthedocs.io/en/stable/modality-specific-files/magnetic-resonance-imaging-data.html#required-gradient-orientation-information)), accessing metadata associated with a data file MUST consider only the applicable file that is lowest in the filesystem hierarchy.